lib/approval_request_listener.rb

```diff
@@ -32,8 +32,17 @@ def subscribe_to_approval_updates
   def process_event(topic)
     approval = ApprovalRequest.find_by!(:approval_request_ref => topic.payload["request_id"])
     approval.order_item.update_message("info", "Task update message received with payload: #{topic.payload}")
+    approval.order_item.update_message("info", "Approval #{approval.id} #{topic.payload['decision']}")
+
     if topic.message == EVENT_REQUEST_FINISHED
       update_and_log_state(approval, topic.payload)
+
+      case approval.order_item
+      when :approved?.to_proc
+        Catalog::SubmitOrder.new(approval.order_item_id).process
```

Comment:
@lindgrenj6 This should be in a separate service object that the listener calls whenever it receives an event. That logic can do all the order item handling and transition it to the ordering from topology. The listener just updates the approval request and calls the service object to say, I just updated your request do what ever is necessary next. Then you would look at how many approval requests are there and if all are approved you send it to topology or a web hook or something else.

Reply:
Added `Catalog::OrderItemTransition`